components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/ExpressionQuery.java

```diff
@@ -242,21 +242,23 @@ protected String createProcessedQuery(String query, InternalParamCollection para
         int count;
         for (int i = 1; i <= paramCount; i++) {
             param = params.getParam(i);
-            value = param.getValue();
-			/*
-			 * value can be null in stored proc OUT params, so it is simply
-			 * treated as a single param, because the number of elements in an
-			 * array cannot be calculated, since there's no actual value passed
-			 * in
-			 */
-            if (value != null && (value.getValueType() == ParamValue.PARAM_VALUE_ARRAY)) {
-                count = (value.getArrayValue()).size();
-            } else {
-                count = 1;
+            if (param != null) {
+                value = param.getValue();
+                /*
+                 * value can be null in stored proc OUT params, so it is simply
+                 * treated as a single param, because the number of elements in an
+                 * array cannot be calculated, since there's no actual value passed
+                 * in
+                 */
+                if (value != null && (value.getValueType() == ParamValue.PARAM_VALUE_ARRAY)) {
+                    count = (value.getArrayValue()).size();
```

Comment:
remove unnecessary ()

Reply:
I have not changed the code inside the for loop due to the null check I added at the top code indentation changed